engine/withdraw_manager.go

```diff
@@ -45,6 +47,12 @@ func (m *WithdrawManager) SubmitWithdrawal(ctx context.Context, req *withdraw.Re
 		RequestDetails: *req,
 	}
 
+	// Determines if the currency can be withdrawn from the exchange
+	errF := exch.CanWithdraw(req.Currency, asset.Spot)
```

Comment:
Any reason why you're making `errF` instead of using `err`?

Reply:
I did it this way because I check for an error ErrCurrencyStateNotFound which is suppressed, further functionality will continue and be error free but it will return this error. So I specifically quarantine this error that has been suppressed so I don't need to nil it afterwards.